### DISCUSSION TITLE
Editor Plugins?

### DISCUSSION THREAD

**Discussion by j531** [migueldeicaza/SwiftGodot#503] [Q&A]
Hello,

Perhaps I've missed it in the documentation, but is there a way to write an editor plugin (`EditorPlugin` subclass) in SwiftGodot?

From what I can see, `plugin.cfg` requires that you declare a GDScript (or C#) based `EditorPlugin` subclass, but I can't work out how to bridge that with a `EditorPlugin` subclass written and exported via SwiftGodot.

Any ideas?

Thanks

**Comment by migueldeicaza** [comment]
It is just a regular class.

I have a branch with a plugin p here, it is very old and outdated, but can give you an idea:

https://github.com/migueldeicaza/SwiftGodot/compare/main...editor-plugin
